### ISSUE TITLE
Add ignoreFunctions: [] to length-zero-no-unit

### ISSUE THREAD

**Issue by zhawkins** [stylelint/stylelint#4945]
> Clearly describe the bug

The `length-zero-no-unit` rule `ignore: ["custom-properties"]` option will only run on custom property declarations like:

```css
a { --x: 0px; }
```

It doesn't handle var functions like:

```
font: var(--x, 0px);
```

I see it being discussed here: https://github.com/stylelint/stylelint/pull/4860#issuecomment-662540528 but don't see a corresponding new issue so I created one. Maybe this would be considered a new feature? 🤔

> Which rule, if any, is the bug related to?

`length-zero-no-unit`

> What code is needed to reproduce the bug?

```css
.foo {
  --x: 0px;
  margin-top: var(--y, 0px);
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "length-zero-no-unit": [
      true,
       {
         "ignore": ["custom-properties"]
       }
    ]
  }
}
```

> Which version of stylelint are you using?

`13.6.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

VScode plugin with `settings.json`

```
    "stylelint.validate": ["scss"],
    "editor.codeActionsOnSave": {
        "source.fixAll.stylelint": true
    }
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

I expect the unit on `0px` within `var(--y, 0px);` to be ignored.

> What actually happened (e.g. what warnings or errors did you get)?

"The following warnings were flagged:"

```shell
   Expected unit (length-zero-no-unit)
```

**Comment by patric-eberle** [comment]
Same issue here. I MUST define a `0px` value as fallback, but the linter triggers.

**Comment by jeddy3** [comment]
@zhawkins Thanks for creating the issue.

I suggest we add an `ignoreFunctions: []` option for this to offer granular control to the user.

For example:

```json
{
  "rules": {
    "length-zero-no-unit": [
      true,
       {
         "ignoreFunctions": ["var", "--custom"]
       }
    ]
  }
}
```

Would allow:

```css
.foo {
  margin-top: var(--y, 0px);
  margin-bottom: --custom(0px);
}
```

I've labelled the issue as ready to implement. @zhawkins or @patric-eberle Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.

**Comment by JuanFML** [comment]
Hi! I was looking at this and I was wondering if I could try to solve this issue as my first PR?

**Comment by hudochenkov** [comment]
@JuanFML of course! It would be great!
